include/mbgl/style/expression/expression.hpp

```diff
@@ -185,18 +185,19 @@ enum class Kind : int32_t {
 
 class Expression {
 public:
-    Expression(Kind kind_, type::Type type_)
+    Expression(Kind kind_, type::Type type_) noexcept
```

Comment:
The move constructor of `type::Type` is not `noexcept`.

Reply:
Good catch, do you have some way to validate these or are you just scanning through them manually?

Reply:
It's safe for a derived class to be noexcept when using a type literal other than `recursive_wrapper<Array>`, right?  e.g., 

```
    Var(std::string name_, std::shared_ptr<Expression> value_) noexcept : Expression(Kind::Var, value_->getType()),
```

Reply:
It's safe to mark a derived class noexcept if it's using a type literal that doesn't involve `recursive_wrapper<Array>`, right?  e.g.:

```
    Any(std::vector<std::unique_ptr<Expression>> inputs_) noexcept
        : Expression(Kind::Any, type::Boolean),
```